test/controllers/beneficial.owner.other.controller.spec.ts

```diff
@@ -1,32 +1,80 @@
 jest.mock("ioredis");
 jest.mock('../../src/controllers/authentication.controller');
+jest.mock('../../src/utils/application.data');
 
+import { natureOfControl, statementCondition, yesNoResponse } from "../../src/model/data.types.model";
+import { BENEFICIAL_OWNER_OTHER_JSON_MOCK, BENEFICIAL_OWNER_OTHER_OBJECT_MOCK } from "../__mocks__/session.mock";
+import { getApplicationData, prepareData, setApplicationData } from "../../src/utils/application.data";
 import { authentication } from "../../src/controllers";
 import { describe, expect, jest, test } from "@jest/globals";
 import request from "supertest";
 import app from "../../src/app";
 import { BENEFICIAL_OWNER_OTHER_URL, MANAGING_OFFICER_URL } from "../../src/config";
 import { NextFunction, Request, Response } from "express";
+import { MESSAGE_ERROR, SERVICE_UNAVAILABLE  } from "../__mocks__/text.mock";
+import { beneficialOwnerOtherType } from '../../src/model';
 
 const mockAuthenticationMiddleware = authentication as jest.Mock;
 mockAuthenticationMiddleware.mockImplementation((req: Request, res: Response, next: NextFunction) => next() );
 
+const mockGetApplicationData = getApplicationData as jest.Mock;
+const mockSetApplicationData = setApplicationData as jest.Mock;
+const mockPrepareData = prepareData as jest.Mock;
+
 const PAGE_TITLE = "Tell us about the corporate beneficial owner";
 
 describe("BENEFICIAL OWNER OTHER controller", () => {
-  test("renders the page", async () => {
-    const resp = await request(app).get(BENEFICIAL_OWNER_OTHER_URL);
 
-    // make some assertions on the response
-    expect(resp.status).toEqual(200);
-    expect(resp.text).toContain(PAGE_TITLE);
+  beforeEach(() => {
+    jest.clearAllMocks();
   });
 
-  test("posts the page and renders the managing-officer page", async () => {
-    const resp = await request(app).post(BENEFICIAL_OWNER_OTHER_URL);
+  describe("GET tests", () => {
+
+    test("renders the page through GET", async () => {
+      mockGetApplicationData.mockReturnValueOnce(BENEFICIAL_OWNER_OTHER_JSON_MOCK);
```

Comment:
mockGetApplicationData should return the APPLICATION_DATA_MOCK, you can add the BENEFICIAL_OWNER_OTHER_OBJECT_MOCK to the APPLICATION_DATA_MOCK and then you don't need the BENEFICIAL_OWNER_OTHER_JSON_MOCK.